plugins/processors/filepath/README.md

```diff
@@ -0,0 +1,207 @@
+* [Rel](https://golang.org/pkg/path/filepath/#Rel) (accessible through `[[processors.filepath.rel]]`)
+* [Dir](https://golang.org/pkg/path/filepath/#Dir) (accessible through `[[processors.filepath.dir]]`)
+* [Clean](https://golang.org/pkg/path/filepath/#Clean) (accessible through `[[processors.filepath.clean]]`)
+* [ToSlash](https://golang.org/pkg/path/filepath/#ToSlash) (accessible through `[[processors.filepath.toslash]]`)
+
+ On top of that, the plugin provides an extra function to retrieve the final path component without its extension. This
+ function is accessible through the `[[processors.filepath.stem]]` configuration item.
+
+Please note that, in this implementation, these functions are processed in the order that they appear above( except for
+`stem` that is applied in the first place).
+
+Specify the `tag` and/or `field` that you want processed in each section and optionally a `dest` if you want the result
+stored in a new tag or field.
+
+If you plan to apply multiple transformations to the same `tag`/`field`, bear in mind the processing order stated above.
+
+## Configuration
+
+```toml
+[[processors.filepath]]
+  ## Treat the tag value as a path and convert it to its last element, storing the result in a new tag
+  # [[processors.filepath.basename]]
+  #   tag = "path"
+  #   dest = "basepath"
+
+  ## Treat the field value as a path and keep all but the last element of path, typically the path's directory
+  # [[processors.filepath.dirname]]
+  #   field = "path"
+
+  ## Treat the tag value as a path, converting it to its the last element without its suffix
+  # [[processors.filepath.stem]]
+  #   tag = "path"
+
+  ## Treat the tag value as a path, converting it to the shortest path name equivalent
+  ## to path by purely lexical processing
+  # [[processors.filepath.clean]]
+  #   tag = "path"
+
+  ## Treat the tag value as a path, converting it to a relative path that is lexically
+  ## equivalent to the source path when joined to 'base_path'
+  # [[processors.filepath.rel]]
+  #   tag = "path"
+  #   base_path = "/var/log"
+
+  ## Treat the tag value as a path, replacing each separator character in path with a '/' character. Has only
+  ## effect on Windows
+  # [[processors.filepath.toslash]]
+  #   tag = "path"
+```
+
+## Considerations
+
+### Clean
+
+Even though `clean` is provided a standalone function, it is also invoked when using the `rel` and `dirname` functions,
+so there is no need to use it along with them.
+
+That is:
+
+ ```toml
+[[processors.filepath]]
+   [[processors.filepath.dir]]
+     tag = "path"
+   [[processors.filepath.clean]]
+     tag = "path"
+ ```
+
+Is equivalent to:
+
+ ```toml
+[[processors.filepath]]
+   [[processors.filepath.dir]]
+     tag = "path"
+ ```
+
+### ToSlash
+
+The effects of this function are only noticeable on Windows platforms, because of the underlying golang implementation.

```

Comment:
It occurs to me that this processor's behavior will depend on which arch is is running on, at least for some of these functions.  I suppose a user can always finish up with ToSlash in order to get consistent results?

Reply:
Yes, the behavior depends on the platform. This can be observed for instance in the `ToSlash` function (https://github.com/golang/go/blob/fffe6229637d1cf37c1400623bdeba57b4a787cb/src/path/filepath/path.go#L165-L170), where the OS separator is used as reference and nothing will be done in case of the separator being `/`. 

Regarding the consistent results yes, one could always use `ToSlash` in case of having different path styles when running on Windows (there is also a `FromSlash` one doing the reverse operation that could also be included in case the user prefers consistent results but with backslash notation).